dips/dip-158.md

```diff
@@ -0,0 +1,775 @@
+|2|InitChargeCommand / InitAuthorizeCommand|Sender|Following customer approval the wallet initializes the payment process and sends the payer data|
+|3|AbortCommand|Sender|Sender decided to abort. Can happen if the sender does not approve merchant data provided by the receiver|
+
+---
+## New Structures
+---
+This section introduces new structures required for P2M scenarios.
+## PaymentInfoObject
+The minimal payment details required in order for the payer to decide whether to accept the payment or reject it. 
+
+Field|Type|Required?|Description|
+|-|-|-|-|
+|receiver|[PaymentReceiverObject](#paymentreceiverobject)|Y|The details of the payment receiver (e.g. merchant)|
+|action|[PaymentActionObject](#paymentactionobject)|Y|Information regarding the type of payment and payment amount|
+|reference_id|str|Y|Unique reference ID of this payment. This value should be globally unique|
+|description|str|N|Description of the payment. To be displayed to the customer|
+
+```
+"payment_info": 
+{
+  "receiver": {
+    PaymentReceiverObject()        
+  },
+  "action": {
+    PaymentActionObject()
+  },
+  "reference_id": "4185027f-0574-6f55-2668-3a38fdb5de98",
+  "description": "payment request description"     
+}
+```
+
+## PaymentReceiverObject
+A PaymentReceiverObject represents the business/merchant in the payment. In P2M scenarios it is the receiver of funds.
+
+|Field|Type|Required?|Description|
+|-|-|-|-|
+| account_address | str | Y | Address of the receiver account. The addresses are encoded using bech32| 
+|business_data|[BusinessDataObject](#businessdataobject)|Y|It includes the details of the merchant to be displayed to the payer|
+
+```
+"receiver": {
+  "account_address": "dm1pllhdmn9m42vcsamx24zrxgs3qqqpzg3ng32kvacapx5fl",         
+  "business_data": {
+    BusinessDataObject()
+  }        
+}
+```
+## PaymentSenderObject
+A PaymentSenderObject represents the person in the payment. In P2M scenarios it is the sender of funds.
+
+|Field|Type|Required?|Description|
+|-|-|-|-|
+| account_address | str | Y | Address of the receiver account. The addresses are encoded using bech32| 
+|payer_data|[PayerDataObject](#payerdataobject)|Y|The details of the payer|
+
+```
+"sender": {
+  "account_address": "dm1pg9q5zs2pg9q5zs2pg9q5zs2pg9skzctpv9skzcg9kmwta",
+  "payer_data": {
+    PayerDataObject() 
+  }          
+}
+```
+
+## PayerDataObject
+The details of the payer. payer_data is the minimal set of payer information required by the merchant to perform risk checks and apply AML policies.  
+|Field|Type|Required?|Description|
+|-|-|-|-|
+|given_name|str|Y|Legal given name of the payer|
+|surname|str|Y|Legal surname of the payer|
+|address|[AddressObject](#addressobject)|Y|Merchant's physical address information|
```

Comment:
Can we change these `address AddressObject`'s to `physical_address? In line with disambiguating from account_address.

Reply:
Renaming address to account_address I think helps with disambiguation.
AddressObject contains the billing details for the customer and the physical address for the merchant.

Reply:
Adding `billing_address` and `physical_address` then would be consistent with `account_address` in that we continue with an unambiguous naming scheme. A first glance at `billing_address` says everything.

Reply:
> |address|[AddressObject](#addressobject)|Y|Merchant's physical address information|

Let's fix the description to show customer's billing address.

Reply:
NP.
I will update in the next version I push to git

Reply:
Eventually decided to leave as is.